src/flownet/config_parser/_config_parser.py

```diff
@@ -480,13 +480,13 @@ def _to_abs_path(path: Optional[str]) -> str:
                 f"The valid phases are 'oil', 'gas', 'water', 'disgas' and 'vapoil'"
             )
     if any(
-        x in [y for y in config.flownet.phases] for x in ["vapoil", "disgas"]
-    ) and not all(x in [y for y in config.flownet.phases] for x in ["oil", "gas"]):
+        x in config.flownet.phases for x in ["vapoil", "disgas"]
+    ) and not all(x in config.flownet.phases for x in ["oil", "gas"]):
```

Comment:
Maybe `set` notation is easier to read here (as the current logic might take some time to understand)?

Not tested, but in that case something like
```python
if not set(["vapoil", "disgas"]).isdisjoint(config.flownet_phases) and not set(["oil", "gas"]).issubset(config.flownet.phases):
```
If so, the other `if` sentences having ~the same structure should also move to `set`.